synapse/http/server.py

```diff
@@ -216,6 +216,14 @@ def render(self, request):
         defer.ensureDeferred(self._async_render_wrapper(request))
         return NOT_DONE_YET
 
+    def _async_render_HEAD(self, request):
+        """
+        The default handling of HEAD method is to treat it as a GET request.
+
+        This is an async version of twisted.web.resource.Resource.render_HEAD.
+        """
+        return self._async_render_GET(request)
```

Comment:
this looks like it will do the wrong thing for resources that do not implement an `_async_render_GET`? it might be better just to special-case HEAD in `_async_render` ?

Reply:
Special case `HEAD` to search for `_async_render_GET`? That might make more sense. 👍 Good catch.

Reply:
yes, exactly

Reply:
Much cleaner. Thanks!